src/mainwindow.cpp

```diff
@@ -122,3 +141,38 @@ void MainWindow::hideLoadingResults()
     ui->openFileProgressBar->hide();
     ui->loadingResultsLabel->hide();
 }
+
+QString MainWindow::formatTimeString(quint64 nanoseconds)
```

Comment:
move this out of the MainWindow class context into an anonymous namespace above the first method implementation in this file. Then first construct temporaries for the integral values and only construct the strings later on as needed

```
namespace {
QString formatTimeString(quint64 nanoseconds)
{
    quint64 totalSeconds = nanoseconds / 1000000000;
    quint64 days = ...;
    quint64 hours = ...;
 
    auto format = [] (quint64 fragment, int precision = 2) -> QString {
        return QString::number(fragment).rightJustified(precision, '0');
    };
    auto optional = [format] (quint64 fragment) -> QString {
        return fragment > 0 ? format(fragment) + QLatin1Char(":") : QString();
    };
    return optional(days) + optional(hours) + optional(minutes) + format(seconds) + QLatin1Char('.') + format(milliseconds, 3);
}
}
```